server/android_notification_server.go

```diff
@@ -220,3 +213,33 @@ func (me *AndroidNotificationServer) SendNotification(msg *PushNotification) Pus
 	}
 	return NewOkPushResponse()
 }
+
+func isRetriableError(err error) bool {
+	// Any error that is not a known error, but the Internal error
+	return !(messaging.IsUnregistered(err) ||
+		messaging.IsInvalidArgument(err) ||
+		messaging.IsQuotaExceeded(err) ||
+		messaging.IsSenderIDMismatch(err) ||
+		messaging.IsThirdPartyAuthError(err))
+}
+
+func (me *AndroidNotificationServer) SendNotificationWithRetry(fcmMsg *messaging.Message, retry int) error {
+	ctx, cancel := context.WithTimeout(context.Background(), me.sendTimeout)
+	defer cancel()
+
+	start := time.Now()
+	_, err := me.client.Send(ctx, fcmMsg)
+	if me.metrics != nil {
+		me.metrics.observerNotificationResponse(PushNotifyAndroid, time.Since(start).Seconds())
+	}
+
+	if isRetriableError(err) {
+		me.logger.Errorf("Failed to send android push did=%v retry=%v error=%v", fcmMsg.Token, retry, err)
+		if nextIteration := retry + 1; nextIteration < MAX_RETRIES {
+			return me.SendNotificationWithRetry(fcmMsg, nextIteration)
```

Comment:
Should we add a wait here before immediately retrying? In the server we do exponential backoff times to give it some time if it's a network hiccup or some sort before continuously retrying - see https://github.com/mattermost/mattermost/blob/master/server/channels/utils/backoff.go#L14-L15

Also in the CWS we use a backoff library that is the same concept https://github.com/mattermost/customer-web-server/blob/master/app/cloud.go#L204